nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java

```diff
@@ -0,0 +1,172 @@
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+import org.osgi.framework.Bundle;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Description of the specific capabilities of a WorkManager for tests, and helper methods.
+ *
+ * @since 10.3
+ */
+public class WorkManagerConfiguration {
+
+    private static final Log log = LogFactory.getLog(WorkManagerConfiguration.class);
+
+    public static final String WORK_MANAGER_PROPERTY = "nuxeo.test.workmanager.implementation";
+
+    public static final String WORK_MANAGER_DEFAULT = "default";
+
+    public static final String WORK_MANAGER_CHRONICLE_STREAM = "chronicle-stream";
+
+    public static final String WORK_MANAGER_KAFKA_STREAM = "kafka-stream";
+
+    private static final String STREAM_WORK_MANAGER_STORESTATE_ENABLED_PROPERTY =
+            "nuxeo.test.stream.workmanager.storestate.enabled";
+
+    private static final String STREAM_WORK_MANAGER_STORESTATE_ENABLED_DEFAULT = "false";
+
+    private String workManagerType;
+
+    private boolean streamWorkManagerStoreStateEnabled;
+
+    final CoreFeature feature;
+
+    public WorkManagerConfiguration(CoreFeature feature) {
+        this.feature = feature;
+    }
+
+    protected void init() {
+        workManagerType = calcWorkManagerType();
+        streamWorkManagerStoreStateEnabled = Boolean.parseBoolean(
+                defaultProperty(STREAM_WORK_MANAGER_STORESTATE_ENABLED_PROPERTY,
+                        STREAM_WORK_MANAGER_STORESTATE_ENABLED_DEFAULT));
+        if (WORK_MANAGER_KAFKA_STREAM.equals(workManagerType)) {
+            initKafka();
+        }
+    }
+
+    protected String calcWorkManagerType() {
+        final String value = System.getProperty(WORK_MANAGER_PROPERTY);
+        if (StringUtils.isNotEmpty(value)) {
+            if (Stream.of(WORK_MANAGER_DEFAULT, WORK_MANAGER_CHRONICLE_STREAM, WORK_MANAGER_KAFKA_STREAM).noneMatch(
+                    s -> s.equals(value))) {
+                throw new NuxeoException(String.format(
+                        "Optional system property '%s' has illegal value '%s'", WORK_MANAGER_PROPERTY, value));
+            }
+            return value;
+        };
+        String workManagerType;
+        if (isKafkaEnabled()) {
+            workManagerType = WORK_MANAGER_KAFKA_STREAM;
+        } else {
+            workManagerType = WORK_MANAGER_DEFAULT;
+        }
+        log.debug(String.format(
+                "Calculated workManagerType value to be '%s' since System Property '%s' was not specified",
+                workManagerType, WORK_MANAGER_PROPERTY));
+        return workManagerType;
+    }
+
+    protected boolean isKafkaEnabled() {
+        return "true".equals(System.getProperty("kafka")) && KafkaUtils.kafkaDetected();
```

Comment:
Please put `"kafka"` in a constant. And could we use something longer like `nuxeo.test.kafka.enabled`, or is this already something used by other parts of the tests?

Reply:
Ok thanks so it's worth extracting that in a constant used in both locations for clarity please.